backend/de.metas.externalreference/src/main/java/de/metas/externalreference/ExternalReferenceRepository.java

```diff
@@ -95,13 +97,21 @@ public ExternalReferenceId save(@NonNull final ExternalReference externalReferen
 	{
 		final I_S_ExternalReference record = InterfaceWrapperHelper.loadOrNew(externalReference.getExternalReferenceId(), I_S_ExternalReference.class);
 
+		final ExternalSystemParentConfigId externalSystemConfigId = (ExternalSystemParentConfigId)externalReference.getExternalSystemParentConfigId(ExternalSystemParentConfigId::ofRepoIdOrNull);
+
 		record.setAD_Org_ID(externalReference.getOrgId().getRepoId());
 		record.setExternalReference(externalReference.getExternalReference());
 		record.setExternalSystem(externalReference.getExternalSystem().getCode());
 		record.setType(externalReference.getExternalReferenceType().getCode());
 		record.setRecord_ID(externalReference.getRecordId());
 		record.setVersion(externalReference.getVersion());
 		record.setExternalReferenceURL(externalReference.getExternalReferenceUrl());
+		record.setExternalSystem_Config_ID(ExternalSystemParentConfigId.toRepoId(externalSystemConfigId));
+
+		if (externalReference.getIsReadOnlyInMetasfresh() != null)
```

Comment:
why updating the record only when `externalReference.getIsReadOnlyInMetasfresh() != null` ?
IMHO you should
* Consider changing S_ExternalReference.IsReadOnlyInMetasfresh column to List (_YesNo) so we will support the null case (btw, is that null case really needed?!)
* or Consider changing ExternalReference.isReadOnlyInMetasfresh from `Boolean` to primitive `boolean`. (not sure but i have the feeling that the right approach)